packages/@uppy/companion/src/server/provider/zoom/index.js

```diff
@@ -34,50 +34,87 @@ class Zoom extends Provider {
-  */
   async list (options) {
     return this.#withErrorHandling('provider.zoom.list.error', async () => {
       const { token } = options
       const query = options.query || {}
-      const { cursor, from, to } = query
       const meetingId = options.directory || ''
+      const requestedYear = query.year ? parseInt(query.year, 10) : null
 
       const client = await getClient({ token })
       const user = await client.get('users/me', { responseType: 'json' }).json()
-
       const { timezone } = user
+      const userTz = timezone || 'UTC'
 
-      if (!from && !to && !meetingId) {
-        const end = cursor && moment.utc(cursor).endOf('day').tz(timezone || 'UTC')
-        return initializeData(user, end)
+      if (meetingId) {
+        const recordingInfo = await client.get(`meetings/${encodeURIComponent(meetingId)}/recordings`, { responseType: 'json' }).json()
+        return adaptData(user, recordingInfo)
       }
 
-      if (from && to) {
-        /*  we need to convert local datetime to UTC date for Zoom query
-        eg: user in PST (UTC-08:00) wants 2020-08-01 (00:00) to 2020-08-31 (23:59)
-        => in UTC, that's 2020-07-31 (16:00) to 2020-08-31 (15:59)
-        */
-        const searchParams = {
-          page_size: PAGE_SIZE,
-          from: moment.tz(from, timezone || 'UTC').startOf('day').tz('UTC').format('YYYY-MM-DD'),
-          to: moment.tz(to, timezone || 'UTC').endOf('day').tz('UTC').format('YYYY-MM-DD'),
+      if (requestedYear) {
+        const yearStartDate = moment.tz({ year: requestedYear, month: 0, day: 1 }, userTz).startOf('day')
+        const yearEndDate = moment.tz({ year: requestedYear, month: 11, day: 31 }, userTz).endOf('day')
+
+        const allMeetingsInYear = []
+        let currentToDate = yearEndDate.clone()
+
+        // Loop backwards in 30-day chunks within the year as Zoom API only allows 30 days at a time
+        while (currentToDate.isSameOrAfter(yearStartDate)) {
+          // Ensure chunk start doesn't go before year start
+          const potentialFromDate = currentToDate.clone().subtract(29, 'days').startOf('day')
+          const currentFromDate = potentialFromDate.isBefore(yearStartDate) ? yearStartDate.clone() : potentialFromDate
+
+          const searchParams = {
+            page_size: PAGE_SIZE,
+            from: currentFromDate.clone().tz('UTC').format('YYYY-MM-DD'), 
+            to: currentToDate.clone().tz('UTC').format('YYYY-MM-DD'),   
+            next_page_token: null, 
+          }
+
+          let currentChunkMeetingsInfo
+          do {
+            if (searchParams.next_page_token === null) delete searchParams.next_page_token
+            currentChunkMeetingsInfo = await client.get('users/me/recordings', { searchParams, responseType: 'json' }).json()
+            if (currentChunkMeetingsInfo.meetings && currentChunkMeetingsInfo.meetings.length > 0) {
+              allMeetingsInYear.push(...currentChunkMeetingsInfo.meetings)
+            }
```

Comment:
no need to check these
```suggestion
            allMeetingsInYear.push(...(currentChunkMeetingsInfo.meetings ?? []))
```